### PR TITLE
Memory leak in hap_send_notification

### PR DESCRIPTION
**Pull request by MacWyznawca**
Memory leak in hap_send_notification (hap_char_update_val)

### CODE REVIEW

File: components/homekit/esp_hap_core/src/esp_hap_ip_services.c [hunk 1]
```diff
@@ -1312,6 +1312,7 @@ static void hap_send_notification(void *arg)
     }
     /* If no characteristic notifications are pending, just exit */
     if (i == 0) {
+	hap_platform_memory_free(char_arr); // Memory leak!
```
Comment (shahpiyushv) [inline components/homekit/esp_hap_core/src/esp_hap_ip_services.c:1315]: Can you change the comment here so that it can be merged as is?

**Comment by shahpiyushv** [comment]
@MacWyznawca , thanks for this fix. I have left a minor comment. Once that is resolved, we can merge this.

File: components/homekit/esp_hap_core/src/esp_hap_ip_services.c (hunk 1 above)
Comment (MacWyznawca) [reply components/homekit/esp_hap_core/src/esp_hap_ip_services.c]: I deleted this comment because everything is rather understandable.

**Comment by shahpiyushv** [comment]
@MacWyznawca , can you make a couple of minor changes?

1. Squash the 2 commits into one
2. Slightly Rephrase commit message

```
Bugfix: Memory leak in hap_send_notification

Fixed a memory leak in hap_send_notification (hap_char_update_val)
```

Normally we import the PR into our internal repo, merge it and then push to GitHub. I could have made these changes myself, but then that would generate a new commit id and not reflect cleanly against this PR :)

**Comment by MacWyznawca** [comment]
Thanks @shahpiyushv ! I think it will be faster if they did it themselves. After this patch, my accessories are working for 10 days without any error or reboot;) Before, after 15-40 minutes I was running out of RAM.

**Comment by shahpiyushv** [comment]
Merged in [this commit](https://github.com/espressif/esp-homekit-sdk/commit/23a2e788295132fd86974cd66686d93b8e3fa5b0).

**Pull request closed without merging**